### PR TITLE
Np 3039/card changes

### PR DESCRIPTION
**Pull request by Nauna**
https://nanopay.atlassian.net/browse/NP-3039

Made CardSelectView Faceted. The Faceted class is in the business repo. https://github.com/nanoPayinc/NANOPAY/pull/11411

TODO The MultiChoiceView - should be named to reflect that its a minMaxChoiceView.
The MultiChoiceVIew renders the CardSelectView of by setting the first choice cls

SORRY for the extra style adjustments.
View first commit for actual changes.

### CODE REVIEW

File: src/foam/u2/view/MultiChoiceView.js [hunk 1]
```diff
@@ -79,16 +80,17 @@ foam.CLASS({
         return [];
       },
       adapt: function(_, n) {
-        if ( ! Array.isArray(n) ) throw new Error("Please submit an array to choices in the MultiChoiceView");
+        if ( ! Array.isArray(n) ) throw new Error('Please submit an array to choices in the MultiChoiceView');
 
-        if ( n.length > 0 ){
+        if ( n.length > 0 ) {
+          n = n.sort();
```
Comment (Nauna) [inline src/foam/u2/view/MultiChoiceView.js:84]: apart of style commit - but this is what is used to alphebetically set the choices of the minMax

File: src/foam/u2/view/MultiChoiceView.js [hunk 2]
```diff
@@ -256,63 +265,63 @@ foam.CLASS({
...
         .end()
         .start()
           .add(
-            self.slot(function(showValidNumberOfChoicesHelper ,isValidNumberOfChoices, choices) {
+            self.slot(function(showValidNumberOfChoicesHelper, isValidNumberOfChoices, choices) {
               return self.E()
               .callIf(! isValidNumberOfChoices && showValidNumberOfChoicesHelper, function() {
                 this
-                  .add("Please select a valid number of choices")
+                  .add('Please select a valid number of choices');
```
Comment (jlhughes) [inline src/foam/u2/view/MultiChoiceView.js]: Assume we need translation?

**Review by jlhughes — COMMENTED** [review]
Looks good.

File: src/foam/u2/view/MultiChoiceView.js [hunk 3]
```diff
@@ -256,63 +265,63 @@ foam.CLASS({
...
               })
               .callIf(isValidNumberOfChoices && showValidNumberOfChoicesHelper, function() {
                 this
                   .start()
-                    .add("Here are your valid selected choices using the outputSelectedChoicesInValueLabelFormat() function:")
+                    .add('Here are your valid selected choices using the outputSelectedChoicesInValueLabelFormat() function:')
```
Comment (kgrgreer) [inline src/foam/u2/view/MultiChoiceView.js]: And here?

File: src/foam/u2/view/MultiChoiceView.js [hunk 4]
```diff
@@ -256,63 +265,63 @@ foam.CLASS({
...
+              });
             })
           )
         .end();
     },
 
     function outputSelectedChoicesInValueLabelFormat() {
       if ( ! this.isValidNumberOfChoices ) {
-        console.warn("Please select a valid number of choices");
+        console.warn('Please select a valid number of choices');
```
Comment (kgrgreer) [inline src/foam/u2/view/MultiChoiceView.js]: Why is this here? / Why is it useful?

File: src/foam/u2/view/MultiChoiceView.js [hunk 5]
```diff
@@ -323,15 +332,16 @@ foam.CLASS({
 
     function outputSelectedChoicesInDAO() {
       if ( ! this.isValidNumberOfChoices ) {
-        console.warn("Please select a valid number of choices");
+        console.warn('Please select a valid number of choices');
```
Comment (kgrgreer) [inline src/foam/u2/view/MultiChoiceView.js]: Why is this here? / Why is it useful?

**Comment by Nauna** [comment]
Comments of @kgrgreer and @jlhughes  made me look at the code I was fixing syntax of. The whole section that was being commented on is now being removed. The section is an error section mixed with what Im assuming is legacy functionality.

Tested in NatureCode setup. All good. None of my testing yesterday, got me there and removal of it does not change the current flow.

More can be done to clean this view. Left a todo - and 2 others are in the code as well. Will ticket a general ticket for phase 2. https://nanopay.atlassian.net/browse/NP-3079

**Comment by jlhughes** [comment]
So just translation remaining.
